contracts/StorageManager.sol

```diff
@@ -155,31 +173,55 @@ contract StorageManager {
     @param provider the provider from which is proposed to take a Offer.
```

Comment:
dev comment needed: if ERC20 token is used, first allow the contract to pull funds.

Reply:
Done